numba/np/arrayobj.py

```diff
@@ -1382,6 +1384,49 @@ def impl(array, shape):
     return impl
 
 
+@register_jitable
+def numpy_broadcast_shapes_list(r, m, shape):
+    for i in range(len(shape)):
+        k = m - len(shape) + i
+        tmp = shape[i]
+        if tmp == 1:
+            continue
+        if r[k] == 1:
+            r[k] = tmp
+        elif r[k] != tmp:
+            raise ValueError("shape mismatch: objects"
+                             " cannot be broadcast"
+                             " to a single shape")
+
+
+@overload(np.broadcast_arrays)
+def numpy_broadcast_arrays(*args):
+    # number of dimensions
+    m = max([arr.ndim for arr in args])
+
+    def impl(*args):
+        # find out the output shape
+        # we can't call np.broadcast_shapes here since args may have arrays
+        # with different shapes and it is not possible to create a list
+        # with those shapes dinamically
+        shape = [1] * m
+        for array in literal_unroll(args):
+            numpy_broadcast_shapes_list(shape, m, array.shape)
+
+        tup = build_int_tuple(m)
```

Comment:
I think the tuple size is based on a compile time constant, such that a change like:
```diff
diff --git a/numba/np/arrayobj.py b/numba/np/arrayobj.py
index da81005..42d8ef5 100644
--- a/numba/np/arrayobj.py
+++ b/numba/np/arrayobj.py
@@ -1404,6 +1404,8 @@ def numpy_broadcast_arrays(*args):
     # number of dimensions
     m = max([arr.ndim for arr in args])
 
+    tup_init = (0,) * m
+
     def impl(*args):
         # find out the output shape
         # we can't call np.broadcast_shapes here since args may have arrays
@@ -1413,8 +1415,7 @@ def numpy_broadcast_arrays(*args):
         for array in literal_unroll(args):
             numpy_broadcast_shapes_list(shape, m, array.shape)
 
-        tup = build_int_tuple(m)
-
+        tup = tup_init
         for i in range(m):
             tup = tuple_setitem(tup, i, shape[i])
```
would make it possible to remove the `build_int_tuple` function entirely.

Reply:
thanks for the tip